metadata-io/src/main/java/com/linkedin/metadata/graph/elastic/ESGraphQueryDAO.java

```diff
@@ -0,0 +1,163 @@
+package com.linkedin.metadata.graph.elastic;
+
+import com.linkedin.metadata.query.Condition;
+import com.linkedin.metadata.query.CriterionArray;
+import com.linkedin.metadata.query.Filter;
+import com.linkedin.metadata.query.RelationshipDirection;
+import com.linkedin.metadata.query.RelationshipFilter;
+import com.linkedin.metadata.utils.elasticsearch.IndexConvention;
+import java.io.IOException;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.reindex.BulkByScrollResponse;
+import org.elasticsearch.index.reindex.DeleteByQueryRequest;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import static com.linkedin.metadata.graph.elastic.ElasticSearchGraphService.*;
+
+
+/**
+ * A search DAO for Elasticsearch backend.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class ESGraphQueryDAO {
+
+  private final RestHighLevelClient client;
+  private final IndexConvention indexConvention;
+
+  /**
+   * Converts {@link CriterionArray} to neo4j query string.
```

Comment:
It says "neo4j query string." here! 

And Neo4j in both the return types.

Reply:
fixed!